wasm/jsapi/instance/caching.any.js

```diff
@@ -0,0 +1,67 @@
+// META: global=jsshell
+// META: script=/wasm/jsapi/wasm-module-builder.js
+
+promise_test(async t => {
+  const builder = new WasmModuleBuilder();
+  builder.addImportedMemory("module", "memory", 0, 128);
+  builder.exportMemoryAs("exported");
+  const buffer = builder.toBuffer()
+
+  const memory = new WebAssembly.Memory({ initial: 0, maximum: 128 });
+  const {instance} = await WebAssembly.instantiate(buffer, {
+    module: { memory }
+  });
+  assert_equals(instance.exports.exported, memory);
+}, "Memory");
+
+promise_test(async t => {
+  const builder = new WasmModuleBuilder();
+  builder.addImportedTable("module", "table", 0, 128);
+  builder.addExportOfKind("exported", kExternalTable, 0);
+  const buffer = builder.toBuffer()
+
+  const table = new WebAssembly.Table({ element: "funcref", initial: 0, maximum: 128 });
+  const {instance} = await WebAssembly.instantiate(buffer, {
+    module: { table }
+  });
+  assert_equals(instance.exports.exported, table);
+}, "Table");
+
+async function exported_fun() {
+  const builder = new WasmModuleBuilder();
+  const index = builder.addImport("module", "fun", kSig_v_v);
+  builder.addExport("exported", index);
+  const buffer = builder.toBuffer()
+
+  const fun = function() {};
+  const {instance} = await WebAssembly.instantiate(buffer, {
+    module: { fun }
+  });
+  return instance.exports.exported;
+}
+
+promise_test(async t => {
+  const builder = new WasmModuleBuilder();
+  const index = builder.addImport("module", "fun", kSig_v_v);
+  builder.addExport("exported", index);
+  const buffer = builder.toBuffer()
+
+  const fun = await exported_fun();
+  const {instance} = await WebAssembly.instantiate(buffer, {
+    module: { fun }
+  });
+  assert_equals(instance.exports.exported, fun);
+}, "Function");
```

Comment:
Should this test ensure caching of a Function or a WebAssembly Instance? Because `exported_fun` is returning an instance?

Reply:
Sorry, I don't understand. The caching only works for Exported Functions, so I need to create one by instantiating a module and grabbing the function from its exports.

(I could export an actual wasm function rather than a wasm-wrapping-js "host function". Not sure if it matters.)

Reply:
Yeah my point was about
> I could export an actual wasm function rather than a wasm-wrapping-js "host function". Not sure if it matters

maybe could we check the identify of the imported JS function too? I didn't understand the multiple levels of WebAssembly at first.